Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -0,0 +1,116 @@
+import defined from "../../Core/defined.js";
+import PrimitiveType from "../../Core/PrimitiveType.js";
+import AttributeType from "../AttributeType.js";
+import VertexAttributeSemantic from "../VertexAttributeSemantic.js";
+import GeometryVS from "../../Shaders/ModelExperimental/GeometryVS.js";
+
+/**
+ * The geometry pipeline stage processes the vertex attributes of a primitive.
+ *
+ * @private
+ */
+export default function GeometryPipelineStage() {}
+
+/**
+ * This pipeline stage processes the vertex attributes of a mesh primitive, adding the attribute declarations to the shaders,
+ * the attribute objects to the render resources and setting the flags as needed.
+ *
+ * Processes a mesh primitive. This stage modifies the following parts of the render resources:
+ * <ul>
+ *  <li> adds attribute and varying declarations for the vertex attributes in the vertex and fragment shaders
+ *  <li> creates the objects required to create VertexArrays
+ *  <li> sets the flag for point primitive types
+ * </ul>
+ *
+ * @param {MeshPrimitiveRenderResources} renderResources The render resources for this mesh primitive.
+ * @param {ModelComponents.Primitive} primitive The mesh primitive.
+ *
+ * @private
+ */
+GeometryPipelineStage.process = function (renderResources, primitive) {
+  var attributeIndex = 0;
+  for (var i = 0; i < primitive.attributes.length; i++) {
+    var attribute = primitive.attributes[i];
+    processAttribute(
+      renderResources,
+      attribute,
+      attribute.semantic === VertexAttributeSemantic.POSITION
+        ? 0
+        : ++attributeIndex
```

Comment:
Personally I don't like seeing `++` embedded in larger expressions. I'd put the increment on its own line.